src/rust/ensogl/lib/core/src/gui/component.rs

```diff
@@ -198,22 +253,45 @@ impl<T> display::Object for ShapeView<T> {
 #[derive(CloneRef,Derivative,Debug,Shrinkwrap)]
 #[derivative(Clone(bound=""))]
 #[allow(missing_docs)]
-pub struct Animation<T> {
+pub struct Animation<T:Animatable> {
     #[shrinkwrap(main_field)]
-    pub simulator : DynSimulator<T>,
+    pub simulator : DynSimulator<T::AnimationSpaceRepr>,
     pub value     : frp::Stream<T>,
 }
 
-impl<T:inertia::Value> Animation<T> {
+#[allow(missing_docs)]
+impl<T:Animatable+Debug+Default+Clone+'static> Animation<T> {
```

Comment:
Are these constraints all needed? I dont get where `'static'`, `Default`, and `Debug` comes from.

Reply:
I think they come from the FRP. I will check that later.

Reply:
Yeah, all of the contsraints are needed for the FRP. Otherwise `T` can't be used in `source::<T>();`

Reply:
If they come from FRP, please use their FRP type alias so we know they are from FRP! Expanding constraitns like this make it error prone in the future - they will change in frp and we will not know where thay came from. Instead, always try to use aliases prepared for it. So just use `T:frp::Data` instead!